contracts/PriorityOperations.sol

```diff
@@ -0,0 +1,295 @@
+        EventQueuePointer memory queuePointer = execResultQueuePointers[_er.strategyId];
+        uint64 aggregateId = queuePointer.tail++;
+        bytes32 ehash = keccak256(
+            abi.encodePacked(_er.strategyId, aggregateId, _er.success, _er.sharesFromBuy, _er.amountFromSell)
+        );
+        pendingExecResults[_er.strategyId][aggregateId] = PendingEvent({
+            ehash: ehash,
+            blockId: uint64(_er.blockLen) - 1, // "pending": baseline of censorship delay
+            status: PendingEventStatus.Pending
+        });
+
+        // Delete pending execution result finalized by this or previous block.
+        while (queuePointer.executeHead < queuePointer.commitHead) {
+            PendingEvent memory pend = pendingExecResults[_er.strategyId][queuePointer.executeHead];
+            if (pend.status != PendingEventStatus.Done || pend.blockId > _er.blockId) {
+                break;
+            }
+            delete pendingExecResults[_er.strategyId][queuePointer.executeHead];
+            queuePointer.executeHead++;
+        }
+        execResultQueuePointers[_er.strategyId] = queuePointer;
+        return aggregateId;
+    }
+
+    /**
+     * @notice add pending epoch update
+     * @param _blockLen number of committed blocks
+     * @return epoch value
+     */
+    function addPendingEpoch(uint256 _blockLen) external onlyController returns (uint64) {
+        uint64 epochId = epochQueuePointer.tail++;
+        uint64 epoch = uint64(block.number);
+        pendingEpochUpdates[epochId] = PendingEpochUpdate({
+            epoch: epoch,
+            blockId: uint64(_blockLen), // "pending": baseline of censorship delay
+            status: PendingEventStatus.Pending
+        });
+        return epoch;
+    }
+
+    /**
+     * @notice Check and update the pending epoch update record.
+     * @param _epoch The epoch value.
+     * @param _blockId Commit block Id.
+     */
+    function checkPendingEpochUpdate(uint64 _epoch, uint256 _blockId) external onlyController {
+        EventQueuePointer memory queuePointer = epochQueuePointer;
+        uint64 epochId = queuePointer.commitHead;
+        require(epochId < queuePointer.tail, ErrMsg.REQ_BAD_EPOCH_TN);
+
+        require(pendingEpochUpdates[epochId].epoch == _epoch, ErrMsg.REQ_BAD_EPOCH);
+        pendingEpochUpdates[epochId].status = PendingEventStatus.Done;
+        pendingEpochUpdates[epochId].blockId = uint64(_blockId); // "done": block holding the transition
+        queuePointer.commitHead++;
+        epochQueuePointer = queuePointer;
+    }
+
+    /**
+     * @notice Dispute if operator failed to reflect an L1-initiated priority tx
+     * in a rollup block within the maxPriorityTxDelay
+     * @param _blockLen number of committed blocks.
+     * @param _maxPriorityTxDelay maximm allowed delay for priority tx
+     */
+    function disputePriorityTxDelay(uint256 _blockLen, uint256 _maxPriorityTxDelay)
```

Comment:
Maybe change to `isPriorityTxDelayViolated`?

Reply:
done